mainboard/LxrStm32/src/Sequencer/EuklidGenerator.c

```diff
@@ -229,10 +238,56 @@ void euklid_rotatePattern(uint8_t length, uint8_t amount)
 
 	euklid_patternBuffer &= ~(0xffff<<length); //Remove "leftovers"
 }
+
+void euklid_setRandomNote( uint8_t trackNr, uint8_t active, uint8_t patternNr )
+{
+	euklid_randomNoteActive[trackNr] = active;
+	euklid_generate(trackNr, patternNr);
+}
+
+// https://stackoverflow.com/questions/1167253/implementation-of-rand 
+unsigned int randomMax (int max)
+{
+   static unsigned int z1 = 12345, z2 = 12345, z3 = 12345, z4 = 12345;
+   unsigned int b;
+   b  = ((z1 << 6) ^ z1) >> 13;
+   z1 = ((z1 & 4294967294U) << 18) ^ b;
+   b  = ((z2 << 2) ^ z2) >> 27; 
+   z2 = ((z2 & 4294967288U) << 2) ^ b;
+   b  = ((z3 << 13) ^ z3) >> 21;
+   z3 = ((z3 & 4294967280U) << 7) ^ b;
+   b  = ((z4 << 3) ^ z4) >> 12;
+   z4 = ((z4 & 4294967168U) << 13) ^ b;
+   return ((z1 ^ z2 ^ z3 ^ z4) / ((double) 4294967168U))*max;
+}
+
 //-----------------------------------------------------
 void euklid_transferPattern(uint8_t trackNr, uint8_t patternNr)
 {
 	uint8_t len=euklid_length[trackNr];
+
+	// hidden behind PAR_EUKLID_NOTE_RANDOM toggle
+	if ( euklid_randomNoteActive[trackNr] > 0 ) {
+		int k;
+		Step *stepPtr;
+
+		for(k=0; k<=127; k++) {
+			uint8_t note = randomMax(127);
+
+			stepPtr=&seq_patternSet.seq_subStepPattern[patternNr][trackNr][k];
+			stepPtr->note 		= note;				// note (--AS was SEQ_DEFAULT_NOTE)
+			
+			stepPtr->param1Nr = PAR_FMAMNT1;
+			stepPtr->param1Val = randomMax(127);
```

Comment:
This doesn't seem to do much. Should add a second option with randomizing parameter locks maybe.

Reply:
These are FM of drumvoice doesn't apply to all

Reply:
skipped for now but mentioned here https://github.com/toban/LXR/issues/2